platform/ios/src/MGLMapView.mm

```diff
@@ -6481,7 +6410,11 @@ - (void)updateScaleBar
     // setting this property.
     if ( ! self.scaleBar.hidden)
     {
+        CGSize originalSize = self.scaleBar.intrinsicContentSize;
         [(MGLScaleBar *)self.scaleBar setMetersPerPoint:[self metersPerPointAtLatitude:self.centerCoordinate.latitude]];
+        if ( ! CGSizeEqualToSize(originalSize, self.scaleBar.intrinsicContentSize)) {
```

Comment:
Have you checked to see if this is called during a camera fly-to?

Reply:
After this PR lands, [let's look at whether we can avoid removing and re-adding the size constraints](https://github.com/mapbox/mapbox-gl-native/issues/14066).